legend-engine-xt-persistence-component/legend-engine-xt-persistence-component-logical-plan/src/main/java/org/finos/legend/engine/persistence/components/planner/NontemporalDeltaPlanner.java

```diff
@@ -237,26 +303,80 @@ public Optional<Condition> getDataSplitInRangeCondition()
+    @Override
+    public Map<StatisticName, LogicalPlan> buildLogicalPlanForPreRunStatistics(Resources resources)
     {
-        if (!deleteIndicatorField.isPresent())
+        Map<StatisticName, LogicalPlan> preRunStatisticsResult = new HashMap<>();
+        if (options().collectStatistics())
         {
-            super.addPostRunStatsForRowsDeleted(postRunStatisticsResult);
+            //Rows deleted
+            addPreRunStatsForRowsDeleted(preRunStatisticsResult);
         }
-        else
+
+        return preRunStatisticsResult;
+    }
+
+    @Override
+    public Map<StatisticName, LogicalPlan> buildLogicalPlanForPostRunStatistics(Resources resources)
+    {
+        Map<StatisticName, LogicalPlan> postRunStatisticsResult = new HashMap<>();
+        if (options().collectStatistics())
         {
-            // Not supported at the moment
+            //Incoming dataset record count
+            addPostRunStatsForIncomingRecords(postRunStatisticsResult);
+            //Rows terminated
+            addPostRunStatsForRowsTerminated(postRunStatisticsResult);
+            //Rows inserted
+            addPostRunStatsForRowsInserted(postRunStatisticsResult);
+            //Rows updated
+            addPostRunStatsForRowsUpdated(postRunStatisticsResult);
+            //Rows deleted
+            if (!this.deleteIndicatorField.isPresent() || !this.deleteIndicatorIsSetCondition.isPresent())
+            {
+                addPostRunStatsForRowsDeleted(postRunStatisticsResult);
+            }
         }
+
+        return postRunStatisticsResult;
     }
 
+    @Override
+    protected void addPreRunStatsForRowsDeleted(Map<StatisticName, LogicalPlan> postRunStatisticsResult)
+    {
+        if (this.deleteIndicatorField.isPresent() && this.deleteIndicatorIsSetCondition.isPresent())
+        {
+            Condition digestMatchCondition = LogicalPlanUtils.getDigestMatchCondition(mainDataset(), stagingDataset(), ingestMode().digestField());
+
+            // Rows Deleted = rows removed (hard-deleted) from sink table
+            LogicalPlan rowsDeletedCountPlan = LogicalPlan.builder().addOps(LogicalPlanUtils
+                .getRecordCount(
+                    stagingDataset(),
+                    ROWS_DELETED.get(),
+                    Optional.of(And.builder().addConditions(this.deleteIndicatorIsSetCondition.get(),
+                        Exists.builder()
+                            .source(Selection.builder()
+                                .source(mainDataset())
+                                .condition(And.builder().addConditions(this.pkMatchCondition, digestMatchCondition).build())
+                                .build())
+                            .build()).build()))).build();
+
+            postRunStatisticsResult.put(ROWS_DELETED, rowsDeletedCountPlan);
+        }
+        else
+        {
+            super.addPreRunStatsForRowsDeleted(postRunStatisticsResult);
+        }
```

Comment:
Maybe we don't need the else block? If we are in the non-delete-indicator mode, this DELETED stats will be handled by the post run stats process (assuming this is the intention here?)

Reply:
removed